pkg/networkmanager/firewall-client.js

```diff
@@ -335,7 +332,12 @@ firewall.addServices = (zones, services) => {
     // We can't use Promise.all() here until cockpit is able to dispatch es2015 promises
     // https://github.com/cockpit-project/cockpit/issues/10956
     // eslint-disable-next-line cockpit/no-cockpit-all
-    return cockpit.all(zones.map(z => services.map(s => firewall.addService(z, s))));
+    return cockpit.all(zones.map(z => services.map(s => firewall.addService(z, s)))).then(function () {
```

Comment:
Style: Seems odd to mix arrow and classic functions here. `() => {` should  work as well?

Also, can you please move the `.then` to the next line, so that this is a bit easier to read?

Reply:
sadly we need the old function form here to have `arguments` defined